### ISSUE TITLE
Rework "connect repository" user interface experience for git/https

### ISSUE THREAD

**Issue by jsoref** [argoproj/argo-cd#19733]
# Summary

Consider https://argocd-stnhqf.bunnyenv.com/settings/repos?addRepo=true
<img width="1157" alt="image" src="https://github.com/user-attachments/assets/a7beee5c-5868-44a1-ae71-01c20a5944db">

0. Already have a repository with the url `https://github.com/agaudreault/argocd-example-apps` (because you're filing a ticket based on experience)
1. Choose your connection method: `VIA HTTPS`
2. Type: `git` (this autofills)
3. Project: _does not autofill_ -- if there's only one current value, it should prefill (there's only one value here: `default`)
4. Project: `default`
5. Repository URL: _does not autofill_ -- this is reasonable
6. Repository URL: `https://github.com/agaudreault/argocd-example-apps`
7. Username (optional): -- unfortunately, Chrome autofilled this field with the credentials for ArgoCD itself -- ideally this field would be keyed such that it could be stored but not confuse password managers with the login for ArgoCD itself
8. Username (optional): empty field
9. Password (optional): -- unfortunately, Chrome autofilled this field with the credentials for ArgoCD itself -- ideally this field would be keyed such that it could be stored but not confuse password managers with the login for ArgoCD itself
10. Password (optional): empty field
11. _There are a number of other fields, but leaving them unchanged is fine._
12. Click the `Connect` button at the top -- on the bright side, it hasn't scrolled offscreen (in many ArgoCD things such buttons do)
    <img width="229" alt="image" src="https://github.com/user-attachments/assets/c2774185-3949-4a1b-b54e-6e4075f9928a">

    > Unable to connect HTTPS repository: existing repository spec is different; use upsert flag to force update; difference in keys "Repo"
    
    Ok, this isn't the failure state i was looking for, but we should talk about it.
    1. Messages about `upsert` are inappropriate here given that there's no such thing exposed on this screen
    2. Why is it giving a message about existing repository specs? Yes, I successfully created a https://github.com/agaudreault/argocd-example-apps.git item, but if it knows it's the same, the error should offer a link to the existing thing instead of being a transient dialog
    3. If the app can know that a value is incompatible with things, it should say so proactively _before_ a user clicks _Save_ or _Connect_ or _Commit_ -- that happens when you enter a username but leave the password field blank:
        <img width="215" alt="image" src="https://github.com/user-attachments/assets/a41c6a59-4a9e-4166-aa91-2ff3647fb8c6">
13. Change the url to include a leading space, e.g. ` https://github.com/argoproj/argo-events` and click Connect -- sometimes this will save ... i have screenshots from when it didn't...
    <img width="889" alt="image" src="https://github.com/user-attachments/assets/ce94ac3b-5429-4115-8540-781d34b47126">
     1. There's no way to edit this failed repository -- the edit option should be in the ellipsis menu -- it's possible to edit the successful repositories by clicking on `Successful` which is odd
     2. There's no way to see in the list view that there's a leading space
     3. If you use the ellipsis and choose disconnect, it's visible in the dialog:
          <img width="686" alt="image" src="https://github.com/user-attachments/assets/1c6155f2-db7a-46fc-bc7a-03d74388ff4d">

# Motivation

I'm trying to set up ArgoCD to demonstrate using ArgoCD. The first step in this appears to be creating a repository (the experience of creating an app without a repository was worse than this, and I'll file it next).

# Proposal

General points:
1. Whitespace that can matter should always be visible.
3. When highlighting things like whitespace, a single quotation character should be used consistently across an application. ArgoCD is inconsistent and uses both `'` and `"` (in addition to not using anything).
4. Whitespace that isn't legal should be rejected up front instead of when/after Creating/Saving.
5. Username fields that are for different kinds of username should be tagged so that they aren't treated as the same -- there are a couple that I can think of -- and ideally each would get a unique id so that browsers can understand they're potentially not the same:
   1. ArgoCD the application
   2. HTTPS credentials for Git
   3. HTTPS credentials for Helm
6. Technical things that don't make sense given the options users see should probably not be shown -- messages should fit the user experience not the underlying API

**Comment by blakepettersson** [comment]
> Project: does not autofill -- if there's only one current value, it should prefill (there's only one value here: default)

That's not true - there is a difference between a repository credential that has an empty project and a non-empty project. A credential with an empty project can be used by any application, whereas if a credential has a project set to `default` it can only be used by applications with a `default` project set.

**Comment by jsoref** [comment]
So, in trying to test some of my more recent PRs, I tripped on this again...

My current thoughts are here: https://cloud-native.slack.com/archives/C01TSERG0KZ/p1726787132736099

None of these should allow spaces:
* namespaces
* urls
* git branches

The only thing that would allow spaces is `paths`.

In a normal html page), I could do something like:
```html
<style>input:invalid{background-color:pink}</style>
<input pattern="\S+">
```

Sample url:
```
data:text/html,<style>input:invalid{background-color:pink}</style><input pattern="\S+">
```
